cardano-node/src/Cardano/Node/Tracing/Tracers.hs

```diff
@@ -650,3 +666,75 @@ mkDiffusionTracersExtra configReflection trBase trForward mbTrEKG _trDataPoint t
        , NonP2P.dtAcceptPolicyTracer = Tracer $
            traceWith dtAcceptPolicyTr
        }
+
+-- FIXME: a more elegant way to 'name' this datapoint?
+instance MetaTrace
+           [(UTCTime, TraceLabelPeer (ConnectionId RemoteAddress)
+                                     (TraceChainSyncClientEvent blk))]
+  where
+  namespaceFor _ = Namespace [] ["ChainSync","Client"]
+
+  severityFor (Namespace _ ["ChainSync","Client"]) _ = Just Info
+  severityFor _ns                                  _ = Nothing
+
+  documentFor  (Namespace _ ["ChainSync","Client"]) =
+    Just "Digest of the ChainSync.Client tracer"
+  documentFor _ns =
+    Nothing
+
+  allNamespaces = [ Namespace [] ["ChainSync","Client"]]
+
+-- FIXME: a more elegant way to 'name' this datapoint?
+instance MetaTrace
+           [(UTCTime, TraceLabelPeer
+                        (ConnectionId RemoteAddress)
+                        (BlockFetch.TraceFetchClientState
+                           (Header blk)))]
+  where
+  namespaceFor _ = Namespace [] ["BlockFetch","Client"]
+
+  severityFor (Namespace _ ["BlockFetch","Client"]) _ = Just Info
+  severityFor _ns                                   _ = Nothing
+
+  documentFor  (Namespace _ ["BlockFetch","Client"]) =
+    Just "Digest of the BlockFetch.Client tracer"
+  documentFor _ns =
+    Nothing
+
+  allNamespaces = [ Namespace [] ["BlockFetch","Client"] ]
+
+
+----------------------------------------------------------------------
+-- Abstraction over Tracer: A Digest creation tracer.
+
+-- | mkDigestTracer p mx - trace the values over the last time period 'p'
+--   with a maximum of max' values.
+
+mkDigestTracer :: NominalDiffTime
+               -> Int
+               -> Trace IO ([(UTCTime, a)])
+               -> IO (Trace IO a)
+mkDigestTracer period max' tr =
+  do
+  traceWith tr []  -- Want this datapoint to immediately have a value.
+  foldMTraceM f [] (contramap unfold tr)
+  where
+  f as _lc a = do
+      now <- getCurrentTime
+      let cutoffTime = addUTCTime (negate period) now 
+      return $ (now,a)
+             : take (max'-1) (takeWhile ((> cutoffTime) . fst) as)
+
+mkDefaultDigestTracer :: Trace IO ([(UTCTime, a)]) -> IO (Trace IO a)
+mkDefaultDigestTracer = mkDigestTracer digestDefaultTime digestDefaultLen
+
+
+-- FIXME: turn these into configuration parameters.
```

Comment:
We can add this to the namespace based configuration, then it can be configured based on namespaces up to individual messages. However, then these digest tracers have to be integrated into the trace-dispatcher library. Currently we have the concept of frequency limiting. This seams to me somehow similar. But up till now its not working together with DataPoints.